src/Aspire.Hosting.Sdk/SDK/Sdk.targets

```diff
@@ -13,34 +15,105 @@
+
+    <!-- First, we assume project is not using Central Package Management, and we try to extract the version
+         from PackageReference Items. -->
+    <ItemGroup>
+      <AppHostPackageReference Include="@(PackageReference)" Condition="'%(Identity)' ==
+  'Aspire.Hosting.AppHost'" />
+    </ItemGroup>
+
+    <!-- Extract the Version metadata -->
+    <PropertyGroup>
+      <AppHostVersion>%(AppHostPackageReference.Version)</AppHostVersion>
+    </PropertyGroup>
+
+    <!-- If the version is still empty, then we assume the project is using Central Package Management, 
+         and we try to extract the PackageVersion Items. -->
+    <ItemGroup Condition="'$(AppHostVersion)' == ''">
+      <AppHostPackageReference />
+      <AppHostPackageReference Include="@(PackageVersion)" Condition="'%(Identity)' ==
+  'Aspire.Hosting.AppHost'" />
+    </ItemGroup>
+
+    <PropertyGroup Condition="'$(AppHostVersion)' == ''">
+      <AppHostVersion>%(AppHostPackageReference.Version)</AppHostVersion>
+      <_UsingCentralPackageManagement>true</_UsingCentralPackageManagement>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(AppHostVersion)' != ''">
+      <__CurrentAppHostVersionMessage> You are using version $(AppHostVersion).</__CurrentAppHostVersionMessage>
+    </PropertyGroup>
+
+    <!-- At this point, we should have the version either by CPM or PackageReference, so we fail if not.  -->
+    <Error Condition="'$(AppHostVersion)' == '' or $([MSBuild]::VersionLessThan('$(AppHostVersion)', '8.2.0'))" 
+    Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that needs a package reference to Aspire.Hosting.AppHost version 8.2.0 or above to work correctly.$(__CurrentAppHostVersionMessage)" />
+
+    <!-- If not using CPM, then directly add the two PackageReferences -->
+    <ItemGroup Condition=" '$(AppHostVersion)' != '' and '$(_UsingCentralPackageManagement)' != 'true' ">
+      <PackageReference Include="Aspire.Dashboard.Sdk.$(NETCoreSdkRuntimeIdentifier)"
+  Version="$(AppHostVersion)" />
+      <PackageReference Include="Aspire.Hosting.Orchestration.$(NETCoreSdkRuntimeIdentifier)"
+  Version="$(AppHostVersion)" />
+    </ItemGroup>
+
+    <!-- If using CPM, then we need to define the PackageVersion items, and then add the versionless PackageReferences
+         to avoid NuGet restore complaining about using CPM but having versioned PackageReferences. -->
+    <ItemGroup Condition=" '$(AppHostVersion)' != '' and '$(_UsingCentralPackageManagement)' == 'true' ">
+      <PackageVersion Include="Aspire.Dashboard.Sdk.$(NETCoreSdkRuntimeIdentifier)"
+  Version="$(AppHostVersion)" />
+      <PackageVersion Include="Aspire.Hosting.Orchestration.$(NETCoreSdkRuntimeIdentifier)"
+  Version="$(AppHostVersion)" />
+
+      <PackageReference Include="Aspire.Dashboard.Sdk.$(NETCoreSdkRuntimeIdentifier)" />
+      <PackageReference Include="Aspire.Hosting.Orchestration.$(NETCoreSdkRuntimeIdentifier)" />
+    </ItemGroup>
```

Comment:
Would using IsImplicitlyDefined=true be an option for these PackageReferences? I believe implicit references are exempted from the CPM PackageVersion requirement (although it's been a while since I tested). It'd potentially allow for simplifying these references.

Reply:
I can test that, but I also don't know what happens if you set that in target files that are not inside the main .NET SDK.

Reply:
Yup, you are right, this works. I'll do that to simplify